gnomad/utils/vcf.py

```diff
@@ -92,3 +340,534 @@ def get_pipe_expr(array_expr: hl.expr.ArrayExpression) -> hl.expr.StringExpressi
+    :param verbose: If True, show top values of annotations being checked, including checks that pass; if False,
+        show only top values of annotations that fail checks.
+    :param subpop: Subpop abbreviation, supplied only if subpopulations are included in the annotation groups being checked.
+    :param sort_order: List containing order to sort label group combinations. Default is SORT_ORDER.
+    :return: None
+    """
+    combo_AC = [ht.info[f"{prefix}AC_{x}"] for x in make_label_combos(label_groups)]
+    combo_AN = [ht.info[f"{prefix}AN_{x}"] for x in make_label_combos(label_groups)]
+    combo_nhomalt = [
+        ht.info[f"{prefix}nhomalt_{x}"] for x in make_label_combos(label_groups)
+    ]
+
+    group = label_groups.pop("group")[0]
+    alt_groups = "_".join(
+        sorted(label_groups.keys(), key=lambda x: sort_order.index(x))
+    )
+
+    annot_dict = {
+        f"sum_AC_{group}_{alt_groups}": hl.sum(combo_AC),
+        f"sum_AN_{group}_{alt_groups}": hl.sum(combo_AN),
+        f"sum_nhomalt_{group}_{alt_groups}": hl.sum(combo_nhomalt),
+    }
+
+    ht = ht.annotate(**annot_dict)
+
+    for subfield in ["AC", "AN", "nhomalt"]:
+        if not subpop:
+            generic_field_check(
+                ht,
+                (
+                    ht.info[f"{prefix}{subfield}_{group}"]
+                    != ht[f"sum_{subfield}_{group}_{alt_groups}"]
+                ),
+                f"{prefix}{subfield}_{group} = sum({subfield}_{group}_{alt_groups})",
+                [
+                    f"info.{prefix}{subfield}_{group}",
+                    f"sum_{subfield}_{group}_{alt_groups}",
+                ],
+                verbose,
+            )
+        else:
+            generic_field_check(
+                ht,
+                (
+                    ht.info[f"{prefix}{subfield}_{group}_{subpop}"]
+                    != ht[f"sum_{subfield}_{group}_{alt_groups}"]
+                ),
+                f"{prefix}{subfield}_{group}_{subpop} = sum({subfield}_{group}_{alt_groups})",
+                [
+                    f"info.{prefix}{subfield}_{group}_{subpop}",
+                    f"sum_{subfield}_{group}_{alt_groups}",
+                ],
+                verbose,
+            )
+
+
+def set_female_y_metrics_to_na(
+    t: Union[hl.Table, hl.MatrixTable],
+) -> Dict[str, hl.expr.Int32Expression]:
+    """
+    Set AC, AN, and nhomalt chrY variant annotations for females to NA (instead of 0).
+
+    :param t: Table/MatrixTable containing female variant annotations.
+    :return: Dictionary with reset annotations
+    """
+    metrics = list(t.row.info)
+    female_metrics = [x for x in metrics if "_female" in x]
+
+    female_metrics_dict = {}
+    for metric in female_metrics:
+        female_metrics_dict.update(
+            {
+                f"{metric}": hl.or_missing(
+                    (~t.locus.contig.in_y_nonpar() | ~t.locus.contig.in_y_par()),
```

Comment:
I am likely just missing something here, but why can't it just be `~(t.locus.contig == 'Y')`?

Reply:
I could change it, but then I'd need to infer the build to get the name of the Y contig. I thought keeping the built in hail functions might be more flexible

Reply:
Ah, yes, that is a good reason